packages/core/src/routes/applications/application.ts

```diff
@@ -262,6 +266,11 @@ export default function applicationRoutes<T extends ManagementApiRouter>(
 
       const { isAdmin, protectedAppMetadata, ...rest } = body;
 
+      const pendingUpdateApplication = await queries.applications.findApplicationById(id);
+      if (pendingUpdateApplication.type === ApplicationType.SAML) {
+        throw new RequestError('application.use_saml_app_api');
+      }
```

Comment:
How about `saml_app_read_only`?

Reply:
Based on the offline discussions, we are now disabling the `GET /applications(/:id)?` API for SAML apps. When we need to modify the console, we will then allow the `GET /applications/(:id)?` API, and at that time, we can change it to `saml_app_read_only`.